components/script/dom/bindings/structuredclone.rs

```diff
@@ -85,15 +110,11 @@ unsafe fn read_blob(cx: *mut JSContext,
 unsafe fn write_blob(blob: Root<Blob>,
                      w: *mut JSStructuredCloneWriter)
                      -> Result<(), ()> {
+    let structured_writer = StructuredCloneWriter{w: w};
     let blob_vec = try!(blob.get_bytes());
-    let blob_length = blob_vec.len();
-    let type_string_bytes = blob.type_string().as_bytes().to_vec();
-    let type_string_length = type_string_bytes.len();
     assert!(JS_WriteUint32Pair(w, StructuredCloneTags::DomBlob as u32, 0));
-    write_length(w, blob_length);
-    write_length(w, type_string_length);
-    assert!(JS_WriteBytes(w, blob_vec.as_ptr() as *const raw::c_void, blob_length));
-    assert!(JS_WriteBytes(w, type_string_bytes.as_ptr() as *const raw::c_void, type_string_length));
+    structured_writer.write_slice(&blob_vec);
+    structured_writer.write_slice(blob.type_string().as_bytes(););
```

Comment:
This doesn't compile as written. We can also make use of the `write_str` method that I propose above using `structured_writer.write_str(&blob.type_string());`.